pkg/resourcemanager/azuresql/azuresqldb/azuresqldb.go

```diff
@@ -108,15 +114,21 @@ func (_ *AzureSqlDbManager) CreateOrUpdateDB(ctx context.Context, resourceGroupN
 			Location:           to.StringPtr(location),
 			DatabaseProperties: &dbProp,
 			Tags:               tags,
+			Sku:                dbSku,
 		})
 
+	if err != nil {
+		return result, err
+	}
+
 	return future.Response(), err
 }
 
 // AddLongTermRetention enables / disables long term retention
 func (_ *AzureSqlDbManager) AddLongTermRetention(ctx context.Context, resourceGroupName string, serverName string, databaseName string, weeklyRetention string, monthlyRetention string, yearlyRetention string, weekOfYear int32) (*http.Response, error) {
 
 	longTermClient, err := azuresqlshared.GetBackupLongTermRetentionPoliciesClient()
+	// TODO: Probably shouldn't return a response at all in the err case here (all through this function)
```

Comment:
I think at times there have been attempts to make the Response more user friendly so we could rely on it for error handling. In the end we have had to just use data embedded in the error types.